megamek/src/megamek/server/totalwarfare/TWGameManager.java

```diff
@@ -15466,28 +15466,38 @@ else if ((entity.heat >= 14) && !entity.isShutDown()) {
                         entity.setShutDown(true);
                     } else {
                         int shutdown = (4 + (((entity.heat - 14) / 4) * 2)) - hotDogMod;
+                        TargetRoll target;
                         if (mtHeat) {
                             shutdown -= 5;
+                            target = new TargetRoll(shutdown, "Base tacops shutdown TN");
                             switch (entity.getCrew().getPiloting()) {
                                 case 0:
                                 case 1:
-                                    shutdown -= 2;
+                                    target.addModifier(-2, "Piloting skill");
                                     break;
                                 case 2:
                                 case 3:
-                                    shutdown -= 1;
+                                    target.addModifier(-1, "Piloting skill");
                                     break;
                                 case 6:
                                 case 7:
-                                    shutdown += 1;
+                                    target.addModifier(+1, "Piloting skill");
                             }
+                        } else {
+                            target = new TargetRoll(shutdown, "Base shutdown TN");
+                        }
+                        boolean riscKit = false;
+                        Mek mek = (Mek) entity;
+                        if (mek.hasRiscHeatSinkOverrideKit()) {
+                            riscKit = true;
```

Comment:
![image](https://github.com/user-attachments/assets/2a2b14a8-f0fe-4a07-ab8a-d98a7bd2dcd5)
this could be rewritten as:
Mek mek = (Mek) entity;
boolean riscKit = mek.hasRiscHeatSinkOverrideKit();
if (riscKit) {

Actually, it isnt necessary to cache hasRiscHeatSinkOverrideKit() as it is already cached in Mek. Which makes me wonder: can the kit be damaged by crit or otherwise be nonfunctional? This is not reflected in the method call.

Reply:
The RISC Kit isn't a piece of equipment, it's an intrinsic property of the 'Mech. Rules-wise it behaves like a quirk, except that CGL wanted it to appear in the Weapons and Equipment Inventory of record sheets. 

It's a lot like the full-head ejection system in that sense.